src/wwwroot/js/genpage/helpers/image_editor.js

```diff
@@ -1860,6 +2081,13 @@ class ImageEditor {
             let [selectX, selectY] = this.imageCoordToCanvasCoord(this.selectX, this.selectY);
             this.drawSelectionBox(selectX, selectY, this.selectWidth * this.zoomLevel, this.selectHeight * this.zoomLevel, this.uiColor, 8 * this.zoomLevel, 0);
         }
+        
+        if (this.tools.shape && this.tools.shape.shapes) {
+            for (let shape of this.tools.shape.shapes) {
+                this.tools.shape.drawShape(shape);
```

Comment:
Shapes should either direct draw, or act as sub-layers on a layer. Doesn't really make sense to render in the main function here like this